src/main/groovy/org/ajoberstar/grgit/auth/UsernamePasswordPassphraseCredentialsProvider.groovy

```diff
@@ -0,0 +1,61 @@
+package org.ajoberstar.grgit.auth
+
+import org.eclipse.jgit.errors.UnsupportedCredentialItem
+import org.eclipse.jgit.transport.CredentialItem
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.URIish
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider
+
+/**
+ * Simple {@link CredentialsProvider} that always uses the same information. Use username, password and passphrase.
+ */
+class UsernamePasswordPassphraseCredentialsProvider extends UsernamePasswordCredentialsProvider {
+	
+	String passphrase;
+	
+	public UsernamePasswordPassphraseCredentialsProvider(String username, String password, String passphrase) {
+		this(username, password.toCharArray(), passphrase);
+	}
+	
+	public UsernamePasswordPassphraseCredentialsProvider(String username, char[] password, String passphrase) {
+		super(username, password);
+		this.passphrase = passphrase;
+	}
+	
+	@Override
+	public boolean supports(CredentialItem... items) {
+		if(! super.supports(items)) {
+			for (CredentialItem i : items) {
+				if (i instanceof CredentialItem.StringType) {
+					CredentialItem.StringType itemString = (CredentialItem.StringType) i;
+					if(itemString.getPromptText() != null && itemString.getPromptText().startsWith("Passphrase")) {

```

Comment:
I'm a little uncomfortable having this based on a string-base match of a user prompt. I traced through some JGit code and found that they wrote this JSch `UserInfo` implementation [CredentialsProviderUserInfo](https://github.com/eclipse/jgit/blob/57a263f1823d164142235a72072154f0568cb61c/org.eclipse.jgit/src/org/eclipse/jgit/transport/CredentialsProviderUserInfo.java) that seems to be the only way you'd get prompted for a SSH passphrase.

I think making our own implementation of `UserInfo` where we can tie into `getPassphrase` to provide JSch the write value might be more solid. We could easily inject our own `UserInfo` in the [JschAgentProxySessionFactory#configure](https://github.com/ajoberstar/grgit/blob/master/src/main/groovy/org/ajoberstar/grgit/auth/JschAgentProxySessionFactory.groovy#L54) method.